tests/protocol/test_connectors.py

```diff
@@ -543,24 +543,44 @@ async def test_sync_done(job, expected_doc_source_update):
 
 @pytest.mark.asyncio
 @pytest.mark.parametrize(
-    "scheduling_enabled, expected_next_sync",
+    "scheduling_enabled, expected_next_sync, job_type",
     [
-        (False, None),
-        (True, mock_next_run),
+        (False, None, JobType.ACCESS_CONTROL),
+        (True, mock_next_run, JobType.ACCESS_CONTROL),
+        (False, None, JobType.FULL),
+        (True, mock_next_run, JobType.FULL),
+        (False, None, JobType.INCREMENTAL),
+        (True, mock_next_run, JobType.INCREMENTAL)
     ],
 )
 @patch("connectors.protocol.connectors.next_run")
-async def test_connector_next_sync(next_run, scheduling_enabled, expected_next_sync):
+async def test_connector_next_sync(next_run, scheduling_enabled, expected_next_sync, job_type):
     connector_doc = {
         "_id": "1",
         "_source": {
-            "scheduling": {"full": {"enabled": scheduling_enabled, "interval": "1 * * * * *"}},
+            "scheduling": {
+                "enabled": scheduling_enabled,
+                "interval": "1 * * * * *",
```

Comment:
related to this comment: https://github.com/elastic/enterprise-search-team/issues/4589#issuecomment-1567909683

Why we want to keep the outer `enabled` and `interval`?

Reply:
We don't need it here, but it'll be kept in the mapping to avoid reindex operations on a migration. See the corresponding discussion inside the implementation doc, but I can remove it from this test 👍

Reply:
I think they are removed from both the [mapping](https://github.com/elastic/ent-search/blob/63ce9682bc92b9d57712e7e267550042f4cabca8/shared_togo/lib/shared_togo/crawler2/connector.rb#L156-L177) and the [migration](https://github.com/elastic/ent-search/blob/63ce9682bc92b9d57712e7e267550042f4cabca8/shared_togo/db/migrate/20230526133312_change_scheduling_and_add_job_type_and_sync_cursor_to_connectors.rb#L61-L69)